addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -433,127 +402,46 @@ function makeActionManager(env) {
-                    // we determine here which actions no longer occur in the nextStack,
-                    // and we manually destroy all their controller's widgets
-                    const nextStackActionIds = nextStack.map((c) => c.action.jsId);
-                    const toDestroy = new Set();
-                    for (const c of controllerStack) {
-                        if (!nextStackActionIds.includes(c.action.jsId)) {
-                            if (c.action.type === "ir.actions.act_window") {
-                                for (const viewType in c.action.controllers) {
-                                    toDestroy.add(c.action.controllers[viewType]);
-                                }
-                            } else {
-                                toDestroy.add(c);
-                            }
-                        }
-                    }
-                    for (const c of toDestroy) {
-                        if (c.exportedState) {
-                            c.exportedState.__legacy_widget__.destroy();
-                        }
-                    }
-                    // END LEGACY CODE COMPATIBILITY
-                    controllerStack = nextStack; // the controller is mounted, commit the new stack
-                    // wait Promise callbacks to be executed
-                    pushState(controller);
-                    browser.sessionStorage.setItem("current_action", action._originalAction);
-                }
-                resolve();
-                env.bus.trigger("ACTION_MANAGER:UI-UPDATED", _getActionMode(action));
-            }
-            willUnmount() {
-                if (action.target === "new" && dialogCloseResolve) {
-                    dialogCloseResolve();
-                }
-            }
-            onHistoryBack() {
-                const previousController = controllerStack[controllerStack.length - 2];
-                if (previousController && !dialogCloseProm) {
-                    restore(previousController.jsId);
-                } else {
-                    _executeCloseAction();
-                }
-            }
-            onTitleUpdated(ev) {
-                controller.title = ev.detail;
-            }
-        }
-        ControllerComponent.template = ControllerComponentTemplate;
-        ControllerComponent.Component = controller.Component;
-
         let nextDialog = {};
+        let removeDialog = null;
         if (action.target === "new") {
             cleanDomFromBootstrap();
+            const { onClose } = dialog;
+            nextDialog = {
+                remove: () => removeDialog(),
+                onClose: onClose || options.onClose,
+            };
             const actionDialogProps = {
                 // TODO add size
-                ActionComponent: ControllerComponent,
-                actionProps: controller.props,
+                ActionComponent: ActionController,
+                actionProps: Object.assign({}, controller.props, {
```

Comment:
I don't like props for the ActionController and those for the real Component being at the same level